src/commands/createWebApp.ts

```diff
@@ -0,0 +1,42 @@
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfigResource } from "azure-arm-website/lib/models";
+import { MessageItem, window } from 'vscode';
+import { IAppCreateOptions } from "vscode-azureappservice";
+import { AzureParentTreeItem, IActionContext, SubscriptionTreeItem } from "vscode-azureextensionui";
+import { WebAppTreeItem } from "../explorer/WebAppTreeItem";
+import { ext } from "../extensionVariables";
+import { deploy } from "./deploy";
+import { setAppCreateOptionsDefault } from "./setAppCreateOptionsDefault";
+
+const yesButton: MessageItem = { title: 'Yes' };
+const noButton: MessageItem = { title: 'No', isCloseAffordance: true };
+
+export async function createWebApp(actionContext: IActionContext, node?: AzureParentTreeItem | undefined): Promise<void> {
+    if (!node) {
+        node = <AzureParentTreeItem>await ext.tree.showTreeItemPicker(SubscriptionTreeItem.contextValue);
+    }
+
+    const createOptions: IAppCreateOptions = await setAppCreateOptionsDefault(actionContext);
+
+    const createdApp = <WebAppTreeItem>await node.createChild(createOptions);
```

Comment:
createChild has known problems passing in createOptions (mostly with the node picker). It seems like you could do `setAppCreateOptionsDefault` directly within `createChildImpl`, which would be ideal